### PR TITLE
[Develop] Porting  to rk3288 board with sdl4.2.3

### PR DESCRIPTION
**Pull request by luwanjia**
1. Remove static libraries of openssl.
2. Update HowToBuild document.
3. Modified some cmake code to build for rk3288
4. Fixed:
    1> fixed: Bluetooth connection not stable
    2> fixed: Crashed some times
    3> fixed: App exception disconnect
    4> fixed: Session ended after reconnect to HMI 3 times
5. Feature:
   1> Add usbmuxd support

### CODE REVIEW

File: src/components/transport_manager/src/transport_manager_impl.cc [hunk 1]
```diff
@@ -627,10 +627,18 @@ void TransportManagerImpl::RemoveConnection(
...
+      if (transport_adapter){
+        if(transport_adapter->GetDeviceType() == transport_manager::transport_adapter::USBMUXD){
+          transport_adapter->RemoveFinalizedConnection(it->device,it->application);
+          connections_.erase(it);
+           break;
+        }
       }
+
       connections_.erase(it);
+      if (transport_adapter) {
```
Comment (BrandonHe) [inline src/components/transport_manager/src/transport_manager_impl.cc]: @ilikegithubcom 
Why here need to check `transport_adapter` again? It's not a good code design,
Please process this condition in the last `if(it->id == id)`

**Review by BrandonHe — APPROVED** [review]

File: src/components/transport_manager/src/transport_manager_impl.cc (hunk 1 above)
Comment (yangbiao01) [reply src/components/transport_manager/src/transport_manager_impl.cc]: 1、Usbmuxd模块为了实现不插拔Usb线，重启App依然可以正常连接，需要这样修改
         transport_adapter->RemoveFinalizedConnection(it->device,it->application);
         connections_.erase(it);
2、在测试usb崩溃的过程中，担心此修改有影响，所以对Usbmuxd模块做了特殊处理，Usbmuxd模块删除后就直接跳出循环了，其他模块还是按照老的流程走。
3、潘工在调试蓝牙的时候，也发现需要这样的修改；
4、卢工提交的develop分支已经合并此修改。

Comment (BrandonHe) [reply src/components/transport_manager/src/transport_manager_impl.cc]: 这里对usbmuxd模块的处理，下面的代码中已经包含，无论`transport_adapter`是什么传输方式，都执行代码：
```
 transport_adapter->RemoveFinalizedConnection(it->device,it->application);
       connections_.erase(it);
       break;
```

Comment (yangbiao01) [reply src/components/transport_manager/src/transport_manager_impl.cc]: 我们这边develop 分支代码已经合并此修改，合并后的代码见下：
void TransportManagerImpl::RemoveConnection(
    const uint32_t id, transport_adapter::TransportAdapter* transport_adapter) {
  LOG4CXX_AUTO_TRACE(logger_);
  LOG4CXX_DEBUG(logger_, "Id: " << id);
  sync_primitives::AutoWriteLock lock(connections_lock_);
  for (std::vector<ConnectionInternal>::iterator it = connections_.begin();
       it != connections_.end();
       ++it) {
    if (it->id == id) {
      if (transport_adapter) {
        transport_adapter->RemoveFinalizedConnection(it->device,it->application);
      }
      connections_.erase(it);
      break;
    }
  }
}

Comment (BrandonHe) [reply src/components/transport_manager/src/transport_manager_impl.cc]: @yangbiao01 
好的，good job

**Comment by BrandonHe** [comment]
Viewed

**Pull request merged**